source/common/router/router.cc

```diff
@@ -691,7 +691,7 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
 
 std::unique_ptr<GenericConnPool>
 Filter::createConnPool(Upstream::ThreadLocalCluster& thread_local_cluster) {
-  GenericConnPoolFactory* factory = nullptr;
+  const GenericConnPoolFactory* factory = nullptr;
```

Comment:
I don't generally expect factories to be const. I think you can drop this qualifier here and in the API to get it.

Reply:
Then we can only get this factory from non-const `Router::Context`. 🤔  If this is acceptable, I don't really have a particular preference and can just drop it.

Reply:
why would it be a problem to get a non-const factory from a non-const context?

Reply:
It's no problem. I just think if we can get it even from const context would be better. But we may use the non-const method of the factory 🤔. I will drop this const qualifier.